src/util/expandThoughts.js

```diff
@@ -32,13 +32,15 @@ const publish = new URLSearchParams(window.location.search).get('publish') != nu
     ...
   }
 */
-export const expandThoughts = (path, thoughtIndex, contextIndex, contextViews = {}, contextChain = [], { depth = 0 } = {}) => {
+export const expandThoughts = (path, thoughtIndex, contextIndex, contextViews = {}, contextChain = [], { depth = 0, distanceFromTableAncestor = -1 } = {}) => {
```

Comment:
Rather than adding data to the recursion and increasing the complexity of the function, just check directly if the grandparent is a table.

```js
const isTableColumn2 = () => attribute(
  contextOf(contextOf(thoughtsRanked)), 
  '=view', 
  { state: { thoughtIndex, contextIndex } }
) === 'Table'
```

Reply:
Sure! I was trying to avoid iteration using recursion logic.

Reply:
`attribute` involves iteration due to `.includes`, but this can be optimized when testing equality. We could create a new function to do it in O(1) like this:

```js
const attributeEquals = (context, attr, value) => {
  const firstChild = getThoughts(context.concat(attr))[0]
  return firstChild && firstChild.value === value
}
```

Reply:
7791ec6b383baa25861aebdccfb2cd5ccf1f620d